pkg/plugins/error.go

```diff
@@ -0,0 +1,21 @@
+package plugins
+
+const (
+	UNSIGNED ErrorCode = iota
+	MODIFIED
+	INVALID
+)
+
+type ErrorCode int
+
+func (e ErrorCode) String() string {
+	return [...]string{"unsigned", "modified", "invalid"}[e]
```

Comment:
I don't think it's safe to assume that `e` starts at 0. Also, what is the purpose of `...`? I tried without, and it's the same result. Never saw that syntax before.

```suggestion
	return []string{"unsigned", "modified", "invalid"}[e]
```

Reply:
This seemed to be a simpler solution than the pattern we seem to follow. For example:

```go
func (p permissionType) String() string {
	names := map[uint8]string{
		uint8(USER):   "user",
		uint8(TEAM):   "team",
		uint8(EDITOR): "editor role",
		uint8(VIEWER): "viewer role",
	}
	return names[uint8(p)]
}
```

It would be safe to assume `e` starts at 0 due to using `iota` as the base for the enum, no?

Aye so the purpose of `...` is the difference between is creating a slice vs an array. I want an array because this list will never change

Reply:
It's not a simpler pattern since it assumes the following, which could break:

* The enum values start at 0 (this _could_ change)
* The enum values have a certain order (this could most definitely change)

Your suggestion just looks more complex than a switch statement. You also have to check if `p` is missing from the map. One benefit of the switch statement is that an analysis tool could tell if you're missing a branch.